Task6.2.3

deseff opened the pull request.

deseff committed:
dot added

deseff committed:
Solution of task from chapter 6.2: Simple calculator (adding and subtracting) with separate test to each operation.

deseff committed:
Removing "<substractingResult>" from line 6.

deseff committed:
Adding RealName to forum SimpleUser. Adding RealName to tests. Emptying TestMain.

deseff committed:
Task 6.2: Test of the class choosing even number form the collection.
Maciek, sth is not working, but I do not know what and why.
Could you please help me :)

Content of the task:

Zadanie: Test klasy wybierającej parzyste liczby z kolekcji
Celem zadania jest przetestowanie klasy, która udostępnia metodę przyjmującą jako argument listę ArrayList obiektów typu Integer, która również zwraca listę (nową) ArrayList obiektów typu Integer, stanowiących podzbiór danych wejściowych z pominięciem liczb nieparzystych.

Brzmi strasznie? Już omawiamy o co chodzi :)

Aby przygotować rozwiązanie zadania:

W katalogu src/main/java utwórz pakiet com.kodilla.testing.collection
W utworzonym pakiecie stwórz klasę OddNumbersExterminator, posiadającą metodę exterminate(ArrayList<Integer> numbers)
przyjmującą jako argument kolekcję ArrayList liczb typu Integer,
która zwraca również kolekcję ArrayList liczb typu Integer, stanowiącą podzbiór kolekcji wejściowej z pominięciem liczb nieparzystych.
W katalogu src/test/java utwórz pakiet com.kodilla.testing.collection.
W utworzonym pakiecie stwórz klasę testową (zbiór testów) CollectionTestSuite, a w niej napisz testy sprawdzające, czy metoda filtrująca liczby nieparzyste działa poprawnie:
testOddNumbersExterminatorEmptyList (sprawdzi, czy klasa zachowuje się poprawnie, gdy lista jest pusta)
testOddNumbersExterminatorNormalList (sprawdzi, czy klasa zachowuje się poprawnie, gdy lista zawiera liczby parzyste i nieparzyste)
Użyj również adnotacji @Before oraz @After, aby wyświetlić informację o tym, jakie operacje (testy) są aktualnie wykonywane.

MARudnicki commented on kodilla-testing/src/test/java/com/kodilla/testing/collection/CollectionTestSuite.java:
```diff
@@ -0,0 +1,57 @@
+            testNumbers.add(i);
+        }
+        //When
+        ArrayList<Integer> resultEvenNumbers = new ArrayList<>(oddNumbersExterminator.exterminate(testNumbers));
```
nie ma potrzeby tworzyc tutaj nowej listy. Wystarczy 
`        ArrayList<Integer> resultEvenNumbers = oddNumbersExterminator.exterminate(testNumbers);`

MARudnicki commented on kodilla-testing/src/main/java/com/kodilla/testing/collection/OddNumbersExterminator.java:
```diff
@@ -0,0 +1,14 @@
+    public ArrayList<Integer> exterminate(ArrayList<Integer> numbers) {
+        ArrayList<Integer> evenNumbers = new ArrayList<Integer>();
+        for(Integer number: numbers) {
+            if(number % 2 == 0);
```
to powinno byc ` if(number % 2 == 0) {   }`

MARudnicki commented on kodilla-testing/src/test/java/com/kodilla/testing/collection/CollectionTestSuite.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    @BeforeClass
+    public void beforeClass() {
```
beforeClass jest uzywane tylko na metodach statycznych

MARudnicki replied:
analogicznie afterClass

MARudnicki commented on kodilla-testing/src/test/java/com/kodilla/testing/collection/CollectionTestSuite.java:
```diff
@@ -0,0 +1,57 @@
+        ArrayList<Integer> resultEvenNumbers = new ArrayList<>(oddNumbersExterminator.exterminate(testNumbers));
+        //Then
+        ArrayList<Integer> expectedList = new ArrayList<Integer>();
+        for(int n = 2; n <= 10; n =+ 2) {
```
`=+` czy `+=` ? To zupenie co innego. Pierwsze jest rownoznaczne z n=2, drugie z n = n + 2

MARudnicki reviewed (changes requested):
Jeszcze uwaga do settings.gradle. Powinno byc jedno w roocie projektu z wpisem o nowym module. Powinno to wygladac
```
rootProject.name = 'kodilla-course'
include 'kodilla-testing'
```

deseff committed:
Task 6.3 (not 6.2)
Correction of the code in classes:
- OddNumbersExterminator
- CollectionTestSuite
Correction in settings.gradle
-------------------------------------------------------------------
Task 6.2: Test of the class choosing even number form the collection.
Maciek, sth is not working, but I do not know what and why.
Could you please help me :)

Content of the task:

Zadanie: Test klasy wybierającej parzyste liczby z kolekcji
Celem zadania jest przetestowanie klasy, która udostępnia metodę przyjmującą jako argument listę ArrayList obiektów typu Integer, która również zwraca listę (nową) ArrayList obiektów typu Integer, stanowiących podzbiór danych wejściowych z pominięciem liczb nieparzystych.

Brzmi strasznie? Już omawiamy o co chodzi :)

Aby przygotować rozwiązanie zadania:

W katalogu src/main/java utwórz pakiet com.kodilla.testing.collection
W utworzonym pakiecie stwórz klasę OddNumbersExterminator, posiadającą metodę exterminate(ArrayList<Integer> numbers)
przyjmującą jako argument kolekcję ArrayList liczb typu Integer,
która zwraca również kolekcję ArrayList liczb typu Integer, stanowiącą podzbiór kolekcji wejściowej z pominięciem liczb nieparzystych.
W katalogu src/test/java utwórz pakiet com.kodilla.testing.collection.
W utworzonym pakiecie stwórz klasę testową (zbiór testów) CollectionTestSuite, a w niej napisz testy sprawdzające, czy metoda filtrująca liczby nieparzyste działa poprawnie:
testOddNumbersExterminatorEmptyList (sprawdzi, czy klasa zachowuje się poprawnie, gdy lista jest pusta)
testOddNumbersExterminatorNormalList (sprawdzi, czy klasa zachowuje się poprawnie, gdy lista zawiera liczby parzyste i nieparzyste)
Użyj również adnotacji @Before oraz @After, aby wyświetlić informację o tym, jakie operacje (testy) są aktualnie wykonywane.
